backend/src/Gitlab/Agent/GitlabCodeReviewerAgent.php

```diff
@@ -0,0 +1,229 @@
+                $tools['getFileContent'] = [
+                    'type' => 'function',
+                    'function' => [
+                        'name' => 'getFileContent',
+                        'description' => 'Get the file content to better understand the changes',
+                        'parameters' => [
+                            'type' => 'object',
+                            'properties' => [
+                                'path' => [
+                                    'type' => 'string',
+                                    'description' => 'The fully qualified path to file which needed to specify to get the file content.',
+                                ],
+                            ],
+                            'required' => ['path'],
+                        ],
+                    ],
+                ];
+            }
+        }
+
+        return array_values($tools);
+    }
+
+    public function getExtraDataFields(): array
+    {
+        return [
+            'reviewCommentPrefix' => [
+                'label' => 'Review comment prefix',
+                'description' => 'e.g. "CodeLlama: ", "GPT-4: ". Usefull to distinguish AI comments from different platforms',
+                'type' => AgentFieldType::STRING,
+                'required' => false,
+            ],
+            'systemPrompt' => [
+                'label' => 'System prompt',
+                'description' => "available tools: getFileContent",
+                'type' => AgentFieldType::TEXT,
+                'required' => true,
+            ],
+            'sampleInput' => [
+                'label' => 'Sample input',
+                'type' => AgentFieldType::TEXT,
+                'required' => false,
+            ],
+            'sampleOutput' => [
+                'label' => 'Sample output',
+                'type' => AgentFieldType::TEXT,
+                'required' => false,
+            ],
+            'aiBaseUrl' => [
+                'label' => 'AI base URL',
+                'description' => 'e.g. "https://api.openai.com/v1", "http://localai/v1"',
+                'type' => AgentFieldType::STRING,
+                'required' => false,
+            ],
+            'gitlabToken' => [
+                'label' => 'GitLab token',
+                'description' => 'GitLab token to access the repository',
+                'type' => AgentFieldType::STRING,
+                'required' => true,
+            ],
+            'maxInputTokens' => [
+                'label' => 'Max input tokens',
+                'description' => 'default: 100000',
+                'type' => AgentFieldType::INT,
+                'required' => false,
+            ],
+            'maxToolCalls' => [
+                'label' => 'Max tool execution rounds',
+                'description' => 'default: 15. Use 0 to disable tool calls.',
+                'type' => AgentFieldType::INT,
+                'required' => false,
+            ],
+        ];
+    }
+}
```

Comment:
BOT4:
The file does not end with a newline. According to POSIX standards and general coding conventions, text files should end with a newline character.